### PR TITLE
[ORPO] use log1p for loss

### PR DESCRIPTION
**Pull request by kashif**
torch.log1p is more accurate for smaller values

### CODE REVIEW

**Review by lewtun — APPROVED** [review]
LGTM! Does this also prevent NaNs appearing in the log odds?

**Review by alvarobartt — APPROVED** [review]
